tools/src/test/java/org/terracotta/utilities/io/FilesStreamCopyTest.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2022 Terracotta, Inc., a Software AG company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.utilities.io;
+
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.CopyOption;
+import java.nio.file.DirectoryNotEmptyException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.time.Duration;
+
+import static java.nio.file.Files.createDirectory;
+import static java.nio.file.Files.createSymbolicLink;
+import static java.nio.file.Files.readAllBytes;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.io.FileMatchers.anExistingDirectory;
+import static org.hamcrest.io.FileMatchers.anExistingFile;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+import static org.junit.internal.matchers.ThrowableMessageMatcher.hasMessage;
+import static org.terracotta.utilities.test.matchers.ThrowsMatcher.threw;
+
+/**
+ * Tests for the {@link Files#copy(InputStream, Path, CopyOption...)} method.
+ */
+public class FilesStreamCopyTest extends FilesTestBase {
+
+  private static final String TEST_RESOURCE_NAME = "/De_finibus_bonorum_et_malorum_Liber_Primus.txt";
```

Comment:
I'll say it because I did it, but it doesn't work anywhere near as well as I expected. Could we try gzip compressing this file to avoid sticking 50k of text in the repo?

Reply:
The repository artifacts are already compressed.  In my file system, it's compressed form is 23,487 bytes.  I needed something that had some size for the test.